File: php/features.php
<?php
declare(strict_types=1);

// AviationweatherData SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AviationweatherDataFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AviationweatherDataBaseFeature();
            case "test":
                return new AviationweatherDataTestFeature();
            default:
                return new AviationweatherDataBaseFeature();
        }
    }
}

File: py/features.py
# AviationweatherData SDK feature factory

from feature.base_feature import AviationweatherDataBaseFeature
from feature.test_feature import AviationweatherDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AviationweatherDataBaseFeature(),
        "test": lambda: AviationweatherDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
